src/client/contexts/RealtimeProvider.jsx

```diff
@@ -82,13 +117,21 @@ export function RealtimeProvider({ children }) {
   }
 
   function sendMessage(messageContent) {
-    socket.emit('send message', { message: messageContent, classroomID: data.classroomMeta.id, });
+    return new Promise((resolve, reject) => {
+      socket.emit('send message', { message: messageContent }, (response) => {
+        if (response.error) reject(response);
+        resolve(response);
+      });
```

Comment:
There are two ways to write a event-emitting function, either use
`socket.emit(eventName, data, callback)` or `socket.emit(eventName, data)`

Using the one with callback means that the client is expecting a response when the server receives this event. The server should respond by:
```js
const [data, callback, meta] = packet;
callback({}) // or put the data you want to return to the client here
```

Either use `emit` without callback, or use `emit` with callback and ensure that the server always respond no matter what (In this case we should use `emit` with callback)